packages/contracts/contracts/Deposit.sol

```diff
@@ -1,234 +1,281 @@
-    require(_olderCheckpoint.stateUpdate.plasmaBlockNumber < _newerCheckpoint.stateUpdate.plasmaBlockNumber);
-
-    // Ensure that the newerCheckpoint has no challenges.
-    bytes32 checkpointId = getCheckpointId(_newerCheckpoint.stateUpdate, _newerCheckpoint.stateUpdate.range);
-    require(checkpoints[checkpointId].outstandingChallenges == 0);
-    
-    // Ensure that the newerCheckpoint is no longer challengeable.
-    require(checkpoints[checkpointId].challengeableUntil < block.number + CHALLENGE_PERIOD);
-
-    // Delete the entries in exits and checkpoints at the olderCheckpointId
-    // delete exits
-    delete checkpoints[checkpointId];
-    delete exitsRedeemableAfter[checkpointId];
-  }
-
-  function challengeCheckpointInvalid(Challenge memory _challenge) private {
-    // TODO
-  }
-
-  function challengeLimboCheckpointAlternateTransaction(
-    bytes32 _limboCheckpoint,
-    bytes memory _alternateTransaction,
-    bytes memory _inclusionProof
-  ) private 
-  {
-    // TODO
-  }
-
-  function removeChallengeCheckpointInvalidHistory(Challenge memory _challenge) private {
-    // TODO 
-  }
-
-  function startExit(Challenge memory _checkpoint, bytes memory _witness) private {
-    // TODO
-  }
-
-  function challengeExitDeprecated(Challenge memory _checkpoint, bytes memory _transaction, bytes memory _inclusionProof) private {
-    // TODO
-  }
-
-  function finalizeExit(bytes32 _exit) private {
-    // TODO
-  }
-  
+     * @notice 
+     * @param _amount TODO
+     * @param _initialState  TODO
+     */
+    function deposit(uint256 _amount, dt.StateObject memory _initialState) public {
+        // Transfer tokens to the deposit contract
+        erc20.transferFrom(msg.sender, address(this), _amount);
+        // Create the Range, StateUpdate & Checkpoint
+        dt.Range memory depositRange = dt.Range({start:totalDeposited, end: totalDeposited + _amount });
+        dt.StateUpdate memory stateUpdate = dt.StateUpdate({
+            range: depositRange, stateObject: _initialState, 
+            depositAddress: address(this), plasmaBlockNumber: getLatestPlasmaBlockNumber() 
+        });
+        dt.Checkpoint memory checkpoint = dt.Checkpoint({
+            stateUpdate: stateUpdate,
+            subrange: depositRange
+        });
+        // Extend depositedRanges & increment totalDeposits
+        extendDepositedRanges(_amount);
+        // Calculate the checkpointId and add it to our finalized checkpoints
+        bytes32 checkpointId = getCheckpointId(checkpoint);
+        CheckpointStatus memory status = CheckpointStatus(
+            {challengeableUntil: block.number + CHALLENGE_PERIOD, outstandingChallenges: 0});
+        checkpoints[checkpointId] = status;
+        // Emit an event which informs us that the checkpoint was finalized
+        emit CheckpointFinalized(checkpointId);
+        emit LogCheckpoint(checkpoint);
+    }
+
+    function extendDepositedRanges(uint256 _amount) public {
```

Comment:
Assuming this shouldn't be public.

Reply:
Yeah this was left public for testing, what do you think the best way to deal with testing is?

Reply:
Ideally you shouldn't need to test private methods (tests should cover things a user is allowed to do). You'd instead want to add extra tests to whatever method makes use of this method.

Reply:
Maybe another option would be pulling out this functionality into another contract with public variables & testing that. Then we can import that contract & store a reference to it and keep that reference private